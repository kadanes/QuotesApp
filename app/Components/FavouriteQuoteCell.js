import React from 'react'
import { View, Text, TouchableHighlight, StyleSheet } from 'react-native'

const FavouriteQuoteCell = (props) => {
    return (
        <TouchableHighlight   
            onPress={() => props.navigation.navigate('Quotes',{id: props.item.parentId, category: props.item.category})}
            style={styles.TableCell}
        >
        <View>
            <Text style={styles.Quote}>{props.item.text}</Text>
            <Text style={styles.Author}>-- {props.item.person}</Text>
            <View style={styles.CategoryPill}>
                <Text style={styles.Category}>
                    {props.item.category}
                </Text>
            </View> 
        </View>
        
        </TouchableHighlight> 
    )
}

export default FavouriteQuoteCell

const styles = StyleSheet.create({
    TableCell: {
        backgroundColor: '#ff6347',
        margin:5,
        padding: 5,
        justifyContent: 'space-around',
        flexDirection: 'column',
        flex: 1 ,
        padding: 10,
        margin: 5,
        borderRadius: 15, 
    },
    "Quote": {
        fontWeight: 'bold',
        color: 'white'
    },
    "Author": {
        fontWeight:'200',
        color:'white',
        justifyContent: 'flex-end',
        alignItems: 'flex-end',
        height: 20
    }, 
    Category: {
        color: '#ff6347',
        fontWeight: 'bold',
        fontSize: 12,
        textTransform: 'capitalize',
        margin: 'auto'
    },
    CategoryPill: {
        marginTop: 10,
        padding: 2,
        height: 20,
        borderRadius: 10,
        backgroundColor: 'white',
        width: 100, 
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center'    
    }
})