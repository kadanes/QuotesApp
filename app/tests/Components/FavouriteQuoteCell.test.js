import React from 'react'
import {shallow} from 'enzyme'

import FavouriteQuoteCell from '../../Components/FavouriteQuoteCell'
import {favouriteItem} from '../fixtures/favourites-fixture'


describe('testing FavouriteQuoteCell', () => {
    let wrapper,navigation 

    beforeEach(() => {
        navigation = {
            navigate: jest.fn()
        }
        wrapper = shallow(<FavouriteQuoteCell navigation={navigation} item={favouriteItem}/>)
    })

    it('should render FavouriteQuoteCell correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('should perform naviggation with correct params',() => {
        wrapper.find('TouchableHighlight').first().props().onPress()
        expect(navigation.navigate).toHaveBeenLastCalledWith(
            'Quotes',
            {
                id: favouriteItem.parentId,
                category: favouriteItem.category
            }
        )
    })
})