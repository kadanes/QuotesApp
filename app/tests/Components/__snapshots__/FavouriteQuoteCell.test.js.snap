// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing FavouriteQuoteCell should render FavouriteQuoteCell correctly 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "#ff6347",
      "borderRadius": 15,
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-around",
      "margin": 5,
      "padding": 10,
    }
  }
  underlayColor="black"
>
  <View>
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      Believe you can and you"re halfway there
    </Text>
    <Text
      style={
        Object {
          "alignItems": "flex-end",
          "color": "white",
          "fontWeight": "200",
          "height": 20,
          "justifyContent": "flex-end",
        }
      }
    >
      -- 
      Theodore Roosevelt
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "flex": 1,
          "height": 20,
          "justifyContent": "center",
          "marginTop": 10,
          "padding": 2,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ff6347",
            "fontSize": 12,
            "fontWeight": "bold",
            "margin": "auto",
            "textTransform": "capitalize",
          }
        }
      >
        inspirational
      </Text>
    </View>
  </View>
</TouchableHighlight>
`;
